commands/sketch/new.go

```diff
@@ -79,9 +83,14 @@ func validateSketchName(name string) error {
 			len(name),
 			sketchNameMaxLength))}
 	}
-	if !sketchNameValidationRegex.MatchString(name) {
-		return &arduino.CantCreateSketchError{Cause: errors.New(tr(`invalid sketch name "%[1]s": the first character must be alphanumeric or "_", the following ones can also contain "-" and ".".`,
+	if !sketchNameValidationRegex.MatchString(name) || strings.HasSuffix(name, ".") {
```

Comment:
Why not change the regexp?

Reply:
I also noticed that it's currently possible to name a sketch `_`. Is it intended to be this way?

Reply:
Starting from the good one:

`^[0-9a-zA-Z_][0-9a-zA-Z_\.-]*[0-9a-zA-Z_-]$ | ^[0-9a-zA-Z_]$` (I've added some spaces for clarity)

you can factor the common prefix `^[0-9a-zA-Z_]` as:

`^[0-9a-zA-Z_] ([0-9a-zA-Z_\.-]*[0-9a-zA-Z_-]$ | $)`

and move the `$` out of the capturing group:

`^[0-9a-zA-Z_] ([0-9a-zA-Z_\.-]*[0-9a-zA-Z_-] | ) $`

also, since we are not interested in the content inside the parenthesis, you can transform the capturing group `( ...... )` to a non-capturing group with `(?: ..... )`

`^[0-9a-zA-Z_] (?:[0-9a-zA-Z_\.-]*[0-9a-zA-Z_-] | ) $`

after removing all the spaces I've added for readability, we have:

`^[0-9a-zA-Z_](?:[0-9a-zA-Z_\.-]*[0-9a-zA-Z_-]|)$`

Reply:
> I also noticed that it's currently possible to name a sketch _. Is it intended to be this way?

A bit weird, but yes, the specification allows it. There are no real reasons to forbid it, actually...

Reply:
Committed your suggestion. Thanks Cristian!